### DISCUSSION TITLE
congrats on 1k starts your cms is awosome

### DISCUSSION THREAD

**Discussion by abdulhakim-alshanqiti** [sveltia/sveltia-cms#276] [General]
i used to use static cms ( a decap/ netlify fork)

and i couldn't get auth to work so i searched google and i found a sveltia cloudflare worker for auth with good step by step guide (was my first cloudflare worker so i needed that thx)

i couldn't use sveltia cms at that time because it was missing the features i needed because it was at the start of this project

from what i saw the project came a long way big thanks to ppl who worked on it

////

also i haven't checked but i assume sveltia doesn't have realtime collaboration support and i was shooked how easy it was with partykit and tiptap and yjs

i updated an example project to have less dependancies and because it couldn't build for me this might help someone who is intrested in this 

https://github.com/kemo-1/partykit-tiptap-yjs/tree/main 

here is the original example 

https://github.com/partykit/partykit-text-editor-example

https://github.com/user-attachments/assets/42bd9b4d-1a10-4420-91be-4025265cb85c

**Comment by abdulhakim-alshanqiti** [comment]
also i wanted to link this project to my friend and i was suprised i didn't star it so i did and apperntly i am the number 1k

**Comment by kyoshino** [comment]
Thanks for the 1,000th star! 🤩 Can’t believe such a small personal project in beta is gaining so much traction.

As for real-time collaboration: Not sure if it will be implemented. It might increase the risk of commit conflicts. For example, in WordPress, when someone is editing an entry, it’s rather locked for other users. I want to see it, but anyway it requires user roles and server-side implementation; it needs to be designed well.

**Reply by abdulhakim-alshanqiti** [reply]
oh yeah the example i linked above uses partykit a js library that deals with all sync related stuff and you can either self host the server or deploy on their platform or deploy to your own cloudflare account


when using yjs its a CDRT ..conflict free replicated data type(don't know what that means honstly) but the idea is whatever you do the user is offline and typed something or changed some settings when clients are back online all of them will see the same thing it will keep everyone changes as best as it can


the example i showed above stores documents in a supabase database (free database has a storage of 500mb)  then when the user makes changes it gets sent to the server and the server merges the changes using Yjs many ppl can collaborate on the same document at the same time

here is a demo i made to share with my friends you can open it in your phone and pc and watch how realtime collaboration works

https://slides.kemo-1.partykit.dev/editor/

(the editor path are for the person who are allowed to edit slides ...there is also the /teacher/ path it needs a trailing Slash at end that path can control the slides of the root path / but not the editor path )

making a new document is as easy as changing one line in the client side to point to the new document name and partykit will deal with creating a new room and syncing changes and saving them

**Reply by kyoshino** [reply]
It wasn’t long ago that Cloudflare acquired PartyKit, and I get the concept. Real-time collaborative editing in Google Docs, Notion, Dropbox Paper, etc. is really fun. I think it can be implemented if Sveltia CMS supports database backends in addition to Git 😆
